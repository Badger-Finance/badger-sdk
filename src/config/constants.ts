import { ARB_CONFIG } from './network/arbitrum.config';
import { AVAX_CONFIG } from './network/avalanche.config';
import { BSC_CONFIG } from './network/binance.config';
import { ETH_CONFIG } from './network/ethereum.config';
import { FTM_CONFIG } from './network/fantom.config';
import { LOCAL_CONFIG } from './network/local.config';
import { MATIC_CONFIG } from './network/polygon.config';

export const ONE_MIN_MS = 60 * 1000;
export const ONE_HOUR_MS = 60 * ONE_MIN_MS;
export const ONE_DAY_MS = 24 * ONE_HOUR_MS;
export const ONE_YEAR_MS = 365 * ONE_DAY_MS;

export const SUPPORTED_NETWORKS = [
  ETH_CONFIG,
  LOCAL_CONFIG,
  BSC_CONFIG,
  MATIC_CONFIG,
  ARB_CONFIG,
  AVAX_CONFIG,
  FTM_CONFIG,
];

export const VERSION = '1.0.77';
