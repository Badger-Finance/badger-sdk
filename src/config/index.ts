export * from './enums';
export * from './types';
export * from './constants';
export * from './interfaces';
export * from './network';
