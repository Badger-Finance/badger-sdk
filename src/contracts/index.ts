/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
export type { BadgerTree } from "./BadgerTree";
export type { Digg } from "./Digg";
export type { Erc20 } from "./Erc20";
export type { Ibbtc } from "./Ibbtc";
export type { Registry } from "./Registry";
export type { RewardsLogger } from "./RewardsLogger";
export type { Sett } from "./Sett";
export type { StrategyV15 } from "./StrategyV15";
export type { VaultV15 } from "./VaultV15";

export { BadgerTree__factory } from "./factories/BadgerTree__factory";
export { Digg__factory } from "./factories/Digg__factory";
export { Erc20__factory } from "./factories/Erc20__factory";
export { Ibbtc__factory } from "./factories/Ibbtc__factory";
export { Registry__factory } from "./factories/Registry__factory";
export { RewardsLogger__factory } from "./factories/RewardsLogger__factory";
export { Sett__factory } from "./factories/Sett__factory";
export { StrategyV15__factory } from "./factories/StrategyV15__factory";
export { VaultV15__factory } from "./factories/VaultV15__factory";
