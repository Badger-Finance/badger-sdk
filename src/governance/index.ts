export { GovernanceService } from './governance.service';
