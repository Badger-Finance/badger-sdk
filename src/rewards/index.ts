export * from './interfaces';
export { RewardsService } from './rewards.service';
