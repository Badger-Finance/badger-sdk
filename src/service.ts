import { providers } from '@0xsequence/multicall';
import { Signer } from 'ethers';

import { BadgerSDK, NetworkConfig } from '.';
import BadgerAPI from './api';
import { Logger } from './logger';

export abstract class Service extends Logger {
  constructor(protected sdk: BadgerSDK) {
    super(sdk.logLevel);
  }

  get provider(): providers.MulticallProvider {
    return this.sdk.provider;
  }

  get signer(): Signer | undefined {
    return this.sdk.signer;
  }

  get connector(): providers.MulticallProvider | Signer {
    return this.address && this.signer ? this.signer : this.provider;
  }

  get config(): NetworkConfig {
    return this.sdk.config;
  }

  get address(): string | undefined {
    return this.sdk.address;
  }

  get api(): BadgerAPI {
    return this.sdk.api;
  }
}
