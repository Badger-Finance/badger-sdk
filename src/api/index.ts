import axios, { AxiosError, AxiosInstance } from 'axios';
import { Network } from '../config/enums/network.enum';
import {
  GasPrices,
  MerkleProof,
  PriceSummary,
  TokenConfiguration,
} from './types';
import {
  Account,
  APIOptions,
  LeaderboardSummary,
  ProtocolMetrics,
  ProtocolSummary,
  Vault,
} from './interfaces';
import { Networkish } from '@ethersproject/networks';
import { NetworkConfig } from '../config/network/network.config';
import { SUPPORTED_NETWORKS } from '../config/constants';
import { Currency } from './enums';

export const DEFAULT_API_URL = 'https://api.badger.com/v2';

export class BadgerAPI {
  private static initialized = false;
  private readonly client: AxiosInstance;
  private network: Network;
  public baseURL: string;

  constructor({
    network = Network.Ethereum,
    baseURL = DEFAULT_API_URL,
  }: APIOptions) {
    if (!BadgerAPI.initialized) {
      for (const config of SUPPORTED_NETWORKS) {
        NetworkConfig.register(config);
      }
      BadgerAPI.initialized = true;
    }

    this.baseURL = baseURL;
    this.client = axios.create({
      baseURL,
    });
    const lookupNetwork = this.isLocal(network) ? Network.Ethereum : network;
    this.network = NetworkConfig.getConfig(lookupNetwork).network;
  }

  async loadPrices(
    currency = Currency.USD,
    network?: Network,
  ): Promise<PriceSummary> {
    return this.get('prices', {
      chain: network ?? this.network,
      currency,
    });
  }

  async loadVaults(
    currency = Currency.USD,
    network?: Network,
  ): Promise<Vault[]> {
    return this.get('vaults', {
      chain: network ?? this.network,
      currency,
    });
  }

  async loadVault(
    address: string,
    currency = Currency.USD,
    network?: Network,
  ): Promise<Vault> {
    return this.get(`vaults/${address}`, {
      chain: network ?? this.network,
      currency,
    });
  }

  async loadSetts(
    currency = Currency.USD,
    network?: Network,
  ): Promise<Vault[]> {
    return this.get('setts', {
      chain: network ?? this.network,
      currency,
    });
  }

  async loadSett(
    address: string,
    currency = Currency.USD,
    network?: Network,
  ): Promise<Vault> {
    return this.get(`setts/${address}`, {
      chain: network ?? this.network,
      currency,
    });
  }

  async loadAccount(address: string, network?: Network): Promise<Account> {
    return this.get(`accounts/${address}`, {
      chain: network ?? this.network,
    });
  }

  async loadTokens(network?: Network): Promise<TokenConfiguration> {
    return this.get('tokens', {
      chain: network ?? this.network,
    });
  }

  async loadProof(address: string, network?: Network): Promise<MerkleProof> {
    return this.get(`proofs/${address}`, {
      chain: network ?? this.network,
    });
  }

  async loadGasPrices(network?: Network): Promise<GasPrices> {
    return this.get('gas', {
      chain: network ?? this.network,
    });
  }

  async loadProtocolMetrics(network?: Network): Promise<ProtocolMetrics> {
    return this.get('metrics', {
      chain: network ?? this.network,
    });
  }

  async loadProtocolSummary(
    currency = Currency.USD,
    network?: Network,
  ): Promise<ProtocolSummary> {
    return this.get('value', {
      chain: network ?? this.network,
      currency,
    });
  }

  async loadLeaderboardSummary(network?: Network): Promise<LeaderboardSummary> {
    return this.get('leaderboards', {
      chain: network ?? this.network,
    });
  }

  private async get<T>(
    path: string,
    params: Record<string, string> = {},
  ): Promise<T> {
    try {
      const { data } = await this.client.get(path, {
        params,
      });
      return data as T;
    } catch (error) {
      const { response } = error as AxiosError;

      if (response) {
        const { status } = response;

        if (status === 404) {
          return {} as T;
        }

        throw new Error(status.toFixed());
      }

      throw error;
    }
  }

  private isLocal(network: Networkish): boolean {
    return network === Network.Local || network === 1337;
  }
}

export * from './enums';
export * from './interfaces';
export * from './types';

export default BadgerAPI;
