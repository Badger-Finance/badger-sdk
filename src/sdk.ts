import { ethers } from 'ethers';
import { providers } from '@0xsequence/multicall';

import { Signer } from '@ethersproject/abstract-signer';
import { Networkish } from '@ethersproject/providers';

import { APIOptions, BadgerAPI } from './api';
import { BadgerGraph } from './graphql';

import { NetworkConfig } from './config/network/network.config';

import { DiggService } from './digg/digg.service';
import { ibBTCService } from './ibbtc/ibbtc.service';
import { RegistryService } from './registry/registry.service';
import { RewardsService } from './rewards/rewards.service';
import { VaultsService } from './vaults/vaults.service';
import { TokensService } from './tokens/tokens.service';

import { SDKProvider } from './config/types/sdk-provider';

import { SUPPORTED_NETWORKS } from './config/constants';

export interface SDKOptions extends APIOptions {
  provider: SDKProvider | string;
}

export class BadgerSDK {
  private static initialized = false;
  private loading: Promise<void>;

  public address?: string;
  public api: BadgerAPI;
  public graph: BadgerGraph;
  public config: NetworkConfig;
  public provider: providers.MulticallProvider;
  public signer?: Signer;

  readonly registry: RegistryService;
  readonly digg: DiggService;
  readonly ibbtc: ibBTCService;
  readonly rewards: RewardsService;
  readonly tokens: TokensService;
  readonly vaults: VaultsService;

  constructor({ network, provider, baseURL }: SDKOptions) {
    if (!BadgerSDK.initialized) {
      for (const config of SUPPORTED_NETWORKS) {
        NetworkConfig.register(config);
      }
      BadgerSDK.initialized = true;
    }

    const sdkProvider = BadgerSDK.getSdkProvider(provider);

    this.provider = new providers.MulticallProvider(sdkProvider);
    this.signer = sdkProvider.getSigner();
    this.loading = this.initialize();
    this.config = NetworkConfig.getConfig(network);
    this.api = new BadgerAPI({
      network: this.config.network,
      baseURL,
    });
    this.graph = new BadgerGraph({
      network: this.config.network,
    });
    this.digg = new DiggService(this);
    this.ibbtc = new ibBTCService(this);
    this.registry = new RegistryService(this);
    this.rewards = new RewardsService(this);
    this.tokens = new TokensService(this);
    this.vaults = new VaultsService(this);
  }

  ready() {
    return Promise.all([this.loading, this.registry.ready(), this.rewards.ready()]);
  }

  update(network: Networkish, provider: SDKProvider) {
    this.updateNetwork(network);
    this.updateProvider(provider);
  }

  updateNetwork(network: Networkish) {
    this.config = NetworkConfig.getConfig(network);
    this.api = new BadgerAPI({
      network: this.config.network,
      baseURL: this.api.baseURL,
    });
    this.graph = new BadgerGraph({
      network: this.config.network,
    });
  }

  updateProvider(provider: SDKProvider) {
    const sdkProvider = BadgerSDK.getSdkProvider(provider);

    this.provider = new providers.MulticallProvider(sdkProvider);
    this.signer = sdkProvider.getSigner();
  }

  private static getSdkProvider(provider: SDKProvider | string): SDKProvider {
    let sdkProvider: SDKProvider;

    if (typeof provider === 'string') {
      sdkProvider = new ethers.providers.JsonRpcBatchProvider(provider);
    } else {
      sdkProvider = provider;
    }

    return sdkProvider;
  }

  private async initialize() {
    try {
      if (this.signer) {
        this.address = await this.signer.getAddress();
      }
    } catch {} // ignore errors from getting address
  }
}
