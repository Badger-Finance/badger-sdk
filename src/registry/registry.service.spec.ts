/* eslint-disable @typescript-eslint/ban-ts-comment */

import { JsonRpcProvider, JsonRpcSigner } from '@ethersproject/providers';
import { BigNumber } from 'ethers';
import { mock, MockProxy } from 'jest-mock-extended';

import { VaultState } from '../api';
import { Network } from '../config';
import { RegistryV2, RegistryV2__factory } from '../contracts';
import { GovernanceService } from '../governance/governance.service';
import { RewardsService } from '../rewards';
import { BadgerSDK } from '../sdk';
import { TEST_ADDR } from '../tests/tests.constants';
import { VaultVersion } from '../vaults';
import { registryAllProductionVaultsMock } from './mocks/registry-all-production-vaults.mock';
import { registryProductionVaultsMock } from './mocks/registry-production-vaults.mock';

describe('RegistryService', () => {
  function prepareSdkMocks(currBlock = 39039987): {
    mockProvider: MockProxy<JsonRpcProvider>;
    registryV2: MockProxy<RegistryV2>;
  } {
    const mockSigner = mock<JsonRpcSigner>();
    mockSigner.getAddress
      .calledWith()
      .mockImplementation(async () => TEST_ADDR);
    const mockProvider = mock<JsonRpcProvider>();
    mockProvider.getSigner.calledWith().mockImplementation(() => mockSigner);
    mockProvider.getBlockNumber
      .calledWith()
      .mockImplementation(async () => currBlock);

    jest.spyOn(RewardsService.prototype, 'ready').mockImplementation();

    const registryV2 = mock<RegistryV2>();

    jest
      .spyOn(RegistryV2__factory, 'connect')
      .mockImplementation(() => registryV2);

    return {
      mockProvider,
      registryV2,
    };
  }

  describe('init and regv2 property a part of sdk', () => {
    let mockProvider: MockProxy<JsonRpcProvider>;
    let registryV2: MockProxy<RegistryV2>;

    beforeEach(async () => {
      const sdkMocks = prepareSdkMocks();

      mockProvider = sdkMocks.mockProvider;
      registryV2 = sdkMocks.registryV2;
    });

    it('should init regestryV2', async () => {
      const sdk = new BadgerSDK({
        network: Network.Ethereum,
        provider: mockProvider,
      });

      await sdk.ready();

      expect(sdk.registry.registry).toBe(registryV2);
    });

    it('should throw an error, if registry not deployed', async () => {
      console.error = jest.fn();

      mockProvider.getCode.mockImplementation(async () => '0x');

      const sdk = new BadgerSDK({
        network: Network.Ethereum,
        provider: mockProvider,
      });

      await sdk.ready();

      expect(sdk.registry.hasRegistry()).toBeFalsy();
      expect(() => sdk.registry.registry).toThrowError();
    });
  });

  describe('methods', () => {
    let sdk: BadgerSDK;
    let registryV2: MockProxy<RegistryV2>;

    beforeEach(async () => {
      const sdkMocks = prepareSdkMocks();

      registryV2 = sdkMocks.registryV2;

      sdk = new BadgerSDK({
        network: Network.Ethereum,
        provider: sdkMocks.mockProvider,
      });

      jest.spyOn(GovernanceService.prototype, 'ready').mockImplementation();

      await sdk.ready();
    });

    describe('get', () => {
      it('should return key value and cache it', async () => {
        const treasuryVaultAddr = '0x45b798384c236ef0d78311D98AcAEc222f8c6F54';

        registryV2.get.mockImplementation(
          async () => '0x45b798384c236ef0d78311D98AcAEc222f8c6F54',
        );

        const keyValue = await sdk.registry.get('treasuryVault');

        expect(keyValue).toBe(treasuryVaultAddr);
        expect(registryV2.get.mock.calls.length).toBe(1);
      });
    });

    describe('keysCount', () => {
      it('should return key value and cache it', async () => {
        const keysCnt = 50;
        registryV2.keysCount.mockImplementation(async () =>
          BigNumber.from(keysCnt),
        );

        const sdkKeyCnt = await sdk.registry.keysCount();

        expect(sdkKeyCnt).toBe(keysCnt);
      });
    });

    describe('getVaults', () => {
      it('should return mapped vaults', async () => {
        registryV2.getVaults.mockImplementation(
          // @ts-ignore
          async () => registryProductionVaultsMock,
        );

        expect(
          await sdk.registry.getVaults(
            VaultVersion.v1_5,
            '0x781E82D5D49042baB750efac91858cB65C6b0582',
          ),
        ).toMatchSnapshot();
      });
    });

    describe('getFilteredProductionVaults', () => {
      it('should return mapped vaults', async () => {
        registryV2.getFilteredProductionVaults.mockImplementation(
          // @ts-ignore
          async () => registryProductionVaultsMock,
        );

        expect(
          await sdk.registry.getFilteredProductionVaults(
            VaultVersion.v1_5,
            VaultState.Open,
          ),
        ).toMatchSnapshot();
      });
    });

    describe('getProductionVaults', () => {
      it('should return mapped vaults and parsed metadata', async () => {
        registryV2.getProductionVaults.mockImplementation(
          // @ts-ignore
          async () => registryAllProductionVaultsMock,
        );

        expect(await sdk.registry.getProductionVaults()).toMatchSnapshot();
      });
    });
  });
});
