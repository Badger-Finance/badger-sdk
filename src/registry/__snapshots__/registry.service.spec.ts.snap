// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegistryService methods getFilteredProductionVaults should return mapped vaults 1`] = `
Array [
  Object {
    "address": "0x96d4dBdc91Bef716eb407e415c9987a9fAfb8906",
    "metadata": Object {
      "behavior": "None",
      "name": "bveOXD",
      "protocol": "0xDAO",
    },
    "state": "open",
    "version": "v1.5",
  },
  Object {
    "address": "0xa8bD8655A0dCABE76913D821Ab437562276b3B59",
    "metadata": Object {
      "behavior": "Ecosystem Helper",
      "name": "boxSOLID",
      "protocol": "0xDAO",
    },
    "state": "open",
    "version": "v1.5",
  },
  Object {
    "address": "0xbF2F3a9ba42A00CA5B18842D8eB1954120e4a2A9",
    "metadata": Object {
      "behavior": "Ecosystem Helper",
      "name": "bveOXD/OXD",
      "protocol": "0xDAO",
    },
    "state": "open",
    "version": "v1.5",
  },
  Object {
    "address": "0xcd1e62b390373fcFeA87Dd06E2497Fdc907935fA",
    "metadata": Object {
      "behavior": "Ecosystem",
      "name": "USDC/DEI",
      "protocol": "0xDAO",
    },
    "state": "open",
    "version": "v1.5",
  },
]
`;

exports[`RegistryService methods getProductionVaults should return mapped vaults and parsed metadata 1`] = `
Array [
  Object {
    "address": "0xFce6e3115D84c94df98BD286C1C67279dc021361",
    "metadata": Object {
      "behavior": "Ecosystem",
      "name": "ðŸ‘» BOO/xBOO",
      "protocol": "Solidex",
    },
    "state": "discontinued",
    "version": "v1",
  },
  Object {
    "address": "0x7cc6049a125388B51c530e51727A87aE101f6417",
    "metadata": Object {
      "behavior": "Ecosystem Helper",
      "name": "WFTM/SEX",
      "protocol": "Solidex",
    },
    "state": "discontinued",
    "version": "v1",
  },
  Object {
    "address": "0xC7cBF5a24caBA375C09cc824481F5508c644dF28",
    "metadata": Object {
      "behavior": "Ecosystem Helper",
      "name": "SOLID/SOLIDsex",
      "protocol": "Solidex",
    },
    "state": "discontinued",
    "version": "v1",
  },
  Object {
    "address": "0x02330E37fcc7C27eB4632b3E31b951fDE5126c48",
    "metadata": Object {
      "behavior": "Bitcoin Rewards",
      "name": "GEIST/g3CRV",
      "protocol": "Solidex",
    },
    "state": "discontinued",
    "version": "v1",
  },
  Object {
    "address": "0x60c5A975Bd0511e45D4C9Ae5592c8CA412BE2310",
    "metadata": Object {
      "behavior": "Ecosystem",
      "name": "ðŸ‘» WFTM/CRV",
      "protocol": "Solidex",
    },
    "state": "discontinued",
    "version": "v1",
  },
  Object {
    "address": "0x561f4dBaFA71ad05694C7ea1b894b9D99f684281",
    "metadata": Object {
      "behavior": "Ecosystem",
      "name": "ðŸ‘» USDC/MIM",
      "protocol": "Solidex",
    },
    "state": "discontinued",
    "version": "v1",
  },
  Object {
    "address": "0x77e803F7d0B6d85731F68CfE1f1CA4c48de105F4",
    "metadata": Object {
      "behavior": "Ecosystem",
      "name": "ðŸ‘» WFTM/SCREAM",
      "protocol": "Solidex",
    },
    "state": "discontinued",
    "version": "v1",
  },
  Object {
    "address": "0x609A51e13fd0896c0c02f0B642756533b29f36Ab",
    "metadata": Object {
      "behavior": "Ecosystem",
      "name": "ðŸ‘» WFTM/renBTC",
      "protocol": "Solidex",
    },
    "state": "discontinued",
    "version": "v1",
  },
  Object {
    "address": "0x3f96f8cA67AcDE9041AD62125C5a4D062b07872c",
    "metadata": Object {
      "behavior": "Ecosystem",
      "name": "ðŸ‘» WFTM/TOMB",
      "protocol": "Solidex",
    },
    "state": "discontinued",
    "version": "v1",
  },
  Object {
    "address": "0xB6D63A4e5Ca740e96C26AdAbcAc73BE78Ee39Dc5",
    "metadata": Object {
      "behavior": "Compounder",
      "name": "WBTC/renBTC",
      "protocol": "Solidex",
    },
    "state": "open",
    "version": "v1",
  },
  Object {
    "address": "0xd9770deC6fdA576450e66f0c441B6b8755F7184A",
    "metadata": Object {
      "behavior": "Compounder",
      "name": "WEVE/USDC",
      "protocol": "Solidex",
    },
    "state": "open",
    "version": "v1",
  },
  Object {
    "address": "0x4145a9B8B7bdC76F14e65c26f9c71e23B9069b25",
    "metadata": Object {
      "behavior": "Bitcoin Rewards",
      "name": "WFTM/CRV",
      "protocol": "Solidex",
    },
    "state": "open",
    "version": "v1",
  },
  Object {
    "address": "0x1AaF7f8154704D80E2380B3DbC967FD0486016e0",
    "metadata": Object {
      "behavior": "Bitcoin Rewards",
      "name": "USDC/MIM",
      "protocol": "Solidex",
    },
    "state": "open",
    "version": "v1",
  },
  Object {
    "address": "0x9a28eeFB2c5F8311f37c39314D78Be85C54B5da6",
    "metadata": Object {
      "behavior": "Bitcoin Rewards",
      "name": "WFTM/renBTC",
      "protocol": "Solidex",
    },
    "state": "open",
    "version": "v1",
  },
  Object {
    "address": "0x96d4dBdc91Bef716eb407e415c9987a9fAfb8906",
    "metadata": Object {
      "behavior": "None",
      "name": "bveOXD",
      "protocol": "0xDAO",
    },
    "state": "open",
    "version": "v1.5",
  },
  Object {
    "address": "0xa8bD8655A0dCABE76913D821Ab437562276b3B59",
    "metadata": Object {
      "behavior": "Ecosystem Helper",
      "name": "boxSOLID",
      "protocol": "0xDAO",
    },
    "state": "open",
    "version": "v1.5",
  },
  Object {
    "address": "0xbF2F3a9ba42A00CA5B18842D8eB1954120e4a2A9",
    "metadata": Object {
      "behavior": "Ecosystem Helper",
      "name": "bveOXD/OXD",
      "protocol": "0xDAO",
    },
    "state": "open",
    "version": "v1.5",
  },
  Object {
    "address": "0xcd1e62b390373fcFeA87Dd06E2497Fdc907935fA",
    "metadata": Object {
      "behavior": "Ecosystem",
      "name": "USDC/DEI",
      "protocol": "0xDAO",
    },
    "state": "open",
    "version": "v1.5",
  },
]
`;

exports[`RegistryService methods getVaults should return mapped vaults 1`] = `
Array [
  Object {
    "address": "0x96d4dBdc91Bef716eb407e415c9987a9fAfb8906",
    "metadata": Object {
      "behavior": "None",
      "name": "bveOXD",
      "protocol": "0xDAO",
    },
    "state": "open",
    "version": "v1.5",
  },
  Object {
    "address": "0xa8bD8655A0dCABE76913D821Ab437562276b3B59",
    "metadata": Object {
      "behavior": "Ecosystem Helper",
      "name": "boxSOLID",
      "protocol": "0xDAO",
    },
    "state": "open",
    "version": "v1.5",
  },
  Object {
    "address": "0xbF2F3a9ba42A00CA5B18842D8eB1954120e4a2A9",
    "metadata": Object {
      "behavior": "Ecosystem Helper",
      "name": "bveOXD/OXD",
      "protocol": "0xDAO",
    },
    "state": "open",
    "version": "v1.5",
  },
  Object {
    "address": "0xcd1e62b390373fcFeA87Dd06E2497Fdc907935fA",
    "metadata": Object {
      "behavior": "Ecosystem",
      "name": "USDC/DEI",
      "protocol": "0xDAO",
    },
    "state": "open",
    "version": "v1.5",
  },
]
`;
